COLLABORATOR_GUIDE.md

```diff
@@ -279,6 +245,150 @@ export default { component: NameOfComponent } as Meta;
+
+(For example, if we abandon Vercel, our Website should still completely work as standalone as possible)
+
+#### What is `next.dynamic.mjs`?
+
+Our whole Website uses a custom renderer for rendering the pages.
+As you might have seen, within the `pages` directory we have [Next.js Dynamic Route](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes) named `[...path].tsx` that matches against all possible routes of the Website.
+
+This means that each `.md(x)` file within `pages/` is not rendered by Next.js regular App Tree (`pages/_document.tsx` and `pages/_app.tsx`) but a custom render tree.
+
+This custom render uses `getStaticPaths` and [Incremental Static Generation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) to generate the full list of supported pages of the Website.
+For example, this allows us to generate Localized Pages for every page that is not translated, by telling Next.js to create a localised path.
+`next.dynamic.mjs` is responsible for getting a full list of the source pages (`pages/en`) and identifying which pages have been translated.
+
+Non-translated pages will have their Localized contexts and translated React message-bags (`react-intl`) but the content will be the same as the source page (English).
+Whereas localized pages will have localized context and content.
+
+This custom solution is also able to decide what paths should be compiled during runtime.
+This is a combination of rules defined on `next.constants.mjs` and `[...path].tsx`.
+
+The `[...path].tsx` file ultimately utilizes the `theme.tsx` file as its layout source.
+This setup enables the loading of the Layout Provider and MDX Provider, which in turn, encapsulate and manage the rendering of any child content or components. This includes both content and components provided by the Layout Provider and the transformed MDX content originating from the `.md(x)` source page.
+
+#### What is `next.data.mjs`?
+
+This file is responsible for loading, fetching and generating build-time required information such as Node.js Release data, Blog Posts Metadata (for pagination and indexation), RSS feeds, etc.
+
+#### What is `site.json`?
+
+This file is used for defining Website Metadata, such as which RSS feeds should be generated, Social Media Information and other Metadata used during the Website build-time.
+We use a JSON format to ease collaboration.
+
+#### What is `next.locales.mjs` and why not use Next.js built-in i18n?
+
+Next.js built-in i18n is great, but it is not flexible enough for our needs. We need to be able to generate a full list of supported locales, and we need to be able to generate a full list of supported pages for each locale. We also need to be able to generate a full list of supported pages for each locale, and we need to be able to generate a full list of supported pages for each locale.
```

Comment:
Looks like we have `full list of supported pages for each locale` 3 times?

Reply:
Lol, didn't even notice that!